library/src/com/google/maps/android/BiMultiMap.java

```diff
@@ -0,0 +1,58 @@
+package com.google.maps.android;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+/**
+ * Extension of HashMap that provides two main features. Firstly it allows reverse lookup
+ * for a key given a value, by storing a second HashMap internally which maps values to keys.
+ * Secondly, it supports Collection values, in which case, each item in the collection is
+ * used as a key in the internal reverse HashMap. It's therefore up to the caller to ensure
+ * the overall set of values, and collection values, are unique.
+ *
+ * Used by GeoJsonRenderer to store GeoJsonFeature instances mapped to corresponding Marker,
+ * Polyline, and Polygon map objects. We want to look these up in reverse to provide access
+ * to GeoJsonFeature instances when map objects are clicked.
+ */
+public class BiMultiMap<K> extends HashMap<K, Object> {
```

Comment:
I think this should be package private.

Also, as this extends HashMap, it should override everything relevant, e.g. clone, clear. Even if just to throw an UnsupportedOperationException. 

It also kinda bugs me that this is using Object everywhere, particularly on a public class (and later in the public methods). Have you considered using 3 separate maps and taking advantage of polymorphism to keep the interface clean?